### DISCUSSION TITLE
Player Stats for Each Week

### DISCUSSION THREAD

**Discussion by agronski** [cwendt94/espn-api#350] [ESPN API]
Have a question about how the stats variable works in the players class.  Trying to use it to pull stats from last season (to create a script to do the same thing for this season) and it looks like using the stats variable you get two dicts back.  One for week 18 (guessing its considered the current week), and one for the entire season. Is there a way to get stats from other weeks throughout the year?  And by stats I mean passing yards, rushing yards etc... not just points.  Newer to using python so maybe it is something simple I'm missing as well

*Edit- after some playing around figured out I needed to grab whatever week I wanted stats for boxscore first, then could see the stats for that week as well as the total

**Comment by cwendt94** [comment]
Yea that is correct! To get the weekly player stats you can use the box_score functionality!

**Reply by cwendt94** [reply]
@agronski If you are looking for just an individual player stat you could also use the player_info functionality. I think it might return the players weekly stats all together instead of having to use box_score.

if not I should be able to tweak the API call to get weekly.

**Reply by agronski** [reply]
Thanks yeah I'm able to get each players weekly stats using the box_score and am able to get what I need.  One other tweak though that looks like it would help though is with the game_played variable.  Would it be difficult to differentiate between a player that hasn't had their game started and one that has?  Right now it is just grouped together and given a value of 0, but would be helpful if there was a way to differentiate between the two

**Reply by cwendt94** [reply]
I will look into it, I can’t remember exactly how that variable is sourced but I think it looks at game time.

**Comment by justin-hsieh** [comment]
@agronski I'm trying to do the same thing that you did. When I specify a certain week in the box_scores call, I still only get week 18 stats. 
I used: 

`player_stats = league.box_scores(12)[0].home_team.roster[0].stats`
`print(player_stats)`

and I get:

`{18: {'projected_points': 17.41, 'projected_breakdown': {'receivingYards': 74.90666011, 'receivingTouchdowns': 0.693150615`...

Could you give an example of what your code looks like to get player stats for a certain week? Thanks in advance.

**Reply by agronski** [reply]
so I'm not sure if this is the intended way to get weekly stats data, but this is what my code looks like (using week 18 as an example.

First I grab the boxscores for that week.

`box_scores = league.box_scores(gameweek)`

then I grab the player I want 

`player = box_scores[0].home_lineup[0]`

Where box_scores[0] grabs the first matchup, and home_lineup[0] grabs the first player of the home team
Then I do:

player_stats = player.stats[18]

Might be an easier way to do it then this but its working for me.

**Reply by justin-hsieh** [reply]
Ahh, I was missing the week number at the end of `.stats`. That's why it kept defaulting to week 18 for me.
Thanks for your help!
